pkg/catalogsourceconfig/configuring.go

```diff
@@ -124,6 +130,27 @@ func (r *configuringReconciler) reconcileCatalogSource(csc *v1alpha1.CatalogSour
 	return nil
 }
 
+// ensurePackages makes sure that the csc's status.PackageMap field is updated
+// at the end of the configuring phase if successful. It iterates over the list of packages
+// and creates a new map of PackageName:Version for each package in the spec.
+func (r *configuringReconciler) ensurePackages(csc *v1alpha1.CatalogSourceConfig) error {
+	newPackageMap := make(map[string]string)
+	packageIDs := csc.GetPackageIDs()
+	for _, packageID := range packageIDs {
+		version, err := r.reader.ReadVersion(packageID)
+		if err != nil {
+			r.log.Errorf("Failed to find package: %v", err)
+			return err
```

Comment:
In theory the only case where this happens is a race condition where an opsrc update deletes the package from the datastore. In that case, we will become eventually consistent since the catalogsourceconfig will get updated on the next sync loop. I actually think it makes sense to set the version to `-1` for now, that way other packages are not stuck in a stale state while we wait for that one to resolve.

Reply:
Updated.